conans/client/conan_command_output.py

```diff
@@ -125,12 +125,12 @@ def _grab_info_data(self, deps_graph, grab_paths):
 
             # Paths
             if isinstance(ref, ConanFileReference) and grab_paths:
-                ref_layout = self._cache.get_ref_layout(self._cache.get_latest_rrev(ref))
+                ref_layout = self._cache.get_ref_layout(ref)
                 item_data["export_folder"] = ref_layout.export()
                 item_data["source_folder"] = ref_layout.source()
                 pref_build_id = build_id(conanfile) or package_id
-                pref_build = self._cache.get_latest_prev(PackageReference(ref, pref_build_id))
-                pref_package = self._cache.get_latest_prev(PackageReference(ref, package_id))
+                pref_build = PackageReference(ref, pref_build_id)
+                pref_package = PackageReference(ref, package_id)
```

Comment:
``ref`` above had recipe revision ``rrev`` but this ``pref`` do NOT have package revision, still the command is returning a "build_folder". This is quite confusing @czoido and probably a cause of the refactor, I think it might be even creating a new temporary folder for a ``prev``?

Reply:
Yes, this fails but the tests for this (`InfoFoldersTest`) have the xfail mark and that's why this is passing green.
This will not create a temporal folder but raise an error because Package revision must be known to get the package layout.